.gitattributes

```diff
@@ -1 +1,3 @@
+* text=auto
+*/vendor/* -text
```

Comment:
I don't really think this is going to work for final new lines, I think it only handles new line style (lf vs crlf).